tests-pytest/action.yml

```diff
@@ -100,13 +100,28 @@ runs:
         python-version: ${{ inputs.python-version }}
         use-cache: ${{ inputs.use-python-cache }}
 
-    - name: "Update pip"
+    - name: "Upgrade pip and disable virtual environment (for poetry only)"
       shell: bash
-      run: python -m pip install -U pip
+      run: |
+        python -m pip install --upgrade pip
+        if grep -q 'build-backend = "poetry\.core\.masonry\.api"' "pyproject.toml"; then
+          # For projects using poetry, do not use a virtual environment.
+          # Store POETRY_VIRTUALENVS_CREATE=false in the GitHub environment to prevent
+          # poetry from creating a virtual environment.
```

Comment:
Maybe just add one more sentence to explain what is the issue with poetry own virtual environment. That will give the full details of this specificity in our action.